tokio/src/io/driver/scheduled_io.rs

```diff
@@ -416,14 +436,20 @@ cfg_io_readiness! {
                 let me = self.get_unchecked_mut();
                 (&me.scheduled_io, &mut me.state, &me.waiter)
             };
-
             loop {
                 match *state {
                     State::Init => {
                         // Optimistically check existing readiness
                         let curr = scheduled_io.readiness.load(SeqCst);
                         let ready = Ready::from_usize(READINESS.unpack(curr));
 
+                        // the information is read via the `curr` in the scheduled io
+                        if ready.is_error() {
+                            return Poll::Ready(Err(
+                                std::io::Error::new(std::io::ErrorKind::Other, "Polling error")
+                            ));
+                        }
```

Comment:
I wonder if we could get a better error for the user here.

Reply:
As in trying to find the underlying cause of the polling error? The issue contained some discussion on that. See https://github.com/tokio-rs/tokio/issues/4349#issuecomment-1004841387 and https://github.com/tokio-rs/tokio/issues/4349#issuecomment-1004851935 

The way I see it this interface would just return something along the lines of "Polling error" and then the user of this interface (who knows what kind of FD they are working with) would do whatever is needed to recover the underlying error (using `SO_ERROR` for example, if dealing with a socket).

I wish there was an `ErrorKind::PollingError` though, but maybe it's just that I don't know enough rust error handling to know how to expose this error to the user of the interface?